cmd/capibmadm/cmd/powervs/port/create.go

```diff
@@ -0,0 +1,136 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package port
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/spf13/cobra"
+
+	client "github.com/IBM-Cloud/power-go-client/clients/instance"
+	"github.com/IBM-Cloud/power-go-client/power/models"
+
+	"sigs.k8s.io/cluster-api/cmd/clusterctl/log"
+
+	"sigs.k8s.io/cluster-api-provider-ibmcloud/cmd/capibmadm/clients/iam"
+	"sigs.k8s.io/cluster-api-provider-ibmcloud/cmd/capibmadm/clients/powervs"
+	"sigs.k8s.io/cluster-api-provider-ibmcloud/cmd/capibmadm/options"
+	"sigs.k8s.io/cluster-api-provider-ibmcloud/cmd/capibmadm/printer"
+	"sigs.k8s.io/cluster-api-provider-ibmcloud/cmd/capibmadm/utils"
+)
+
+type portCreateOptions struct {
+	networkID   string
+	ipAddress   string
+	description string
+}
+
+// CreateCommand create powervs network port.
+func CreateCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "create",
+		Short: "Create PowerVS Port",
+		Example: `
+# Create PowerVS network port
+export IBMCLOUD_API_KEY=<api-key>
+capibmadm powervs port create --network <netword-id/network-name> --description <description> --service-instance-id <service-instance-id> --zone <zone>`,
+	}
+
+	var portCreateOption portCreateOptions
+	cmd.Flags().StringVar(&portCreateOption.networkID, "network", "", "Network ID/ name on which port is to be created (Required)")
+	cmd.Flags().StringVar(&portCreateOption.ipAddress, "ip-address", "", "IP Address to be assigned to the port")
+	cmd.Flags().StringVar(&portCreateOption.description, "description", "", "Description of the port")
+	cmd.RunE = func(cmd *cobra.Command, args []string) error {
+		if err := createPort(cmd.Context(), portCreateOption); err != nil {
+			return err
+		}
+		return nil
+	}
+	options.AddCommonFlags(cmd)
+	_ = cmd.MarkFlagRequired("network")
+	return cmd
+}
+
+func createPort(ctx context.Context, portCreateOption portCreateOptions) error {
+	logger := log.Log
+	logger.Info("Creating Port ", "Network ID/Name", portCreateOption.networkID, "IP Address", portCreateOption.ipAddress, "Description", portCreateOption.description, "service-instance-id", options.GlobalOptions.ServiceInstanceID, "zone", options.GlobalOptions.PowerVSZone)
+	accountID, err := utils.GetAccountID(ctx, iam.GetIAMAuth())
+	if err != nil {
+		return err
+	}
+	session, err := powervs.NewPISession(accountID, options.GlobalOptions.PowerVSZone, options.GlobalOptions.Debug)
+	if err != nil {
+		return err
+	}
+
+	networkClient := client.NewIBMPINetworkClient(ctx, session, options.GlobalOptions.ServiceInstanceID)
+	networks, err := networkClient.GetAll()
```

Comment:
@Shilpa-Gokul `Get()` function seems to work with both name/Id fine, so you wont have to do additional work. You can refer [this PR](https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/pull/1084) for more info and re-tweak your code accordingly.